packages/dashboard/src/components/DashboardFilters/DashboardFiltersEditor/components/FilterEditor.tsx

```diff
@@ -77,21 +80,55 @@ const FilterEditor: React.FC<FilterEditorProps> = ({ filter, config, updateFilte
       }
 
       if (_dataSet.data) {
-        _dataSet.data.forEach(row => {
-          Object.keys(row).forEach(columnName => {
-            columns[columnName] = true
-          })
-        })
+        callback(_dataSet.data)
       }
     }
+  }
+
+  const loadValuesforColumn = async () => {
+    let values = []
+
+    await iterateDatasets(dataset => {
+      values = values.concat(_.uniq(dataset.map(row => row[filter.columnName])))
+    })
+
+    sortByOrderedValues(values, filter)
+    setColumnValues(values)
+  }
+
+  const loadColumnData = async () => {
+    const columns = {}
+
+    await iterateDatasets(dataset => {
+      dataset.forEach(row => {
+        Object.keys(row).forEach(columnName => {
+          columns[columnName] = true
+        })
+      })
+    })
 
     setColumns(Object.keys(columns))
   }
 
+  const handleFilterOrder = (sourceIndex, destinationIndex) => {
+    let orderedValues = [...columnValues]
+
+    let placeholder = orderedValues[sourceIndex]
+    orderedValues[sourceIndex] = orderedValues[destinationIndex]
+    orderedValues[destinationIndex] = placeholder
+    
+    updateFilterProp('orderedValues', orderedValues)
+    setColumnValues(orderedValues)
+  }
+
   useEffect(() => {
     loadColumnData()
   }, [config.datasets])
 
+  useEffect(() => {
+    loadValuesforColumn()

```

Comment:
rather than creating a useEffect, make a function that calls loadValuesForColumn declaratively. you already know we want to take an action when the onChange event happens for the columnName, we don't need to wait a whole render cycle to do something about it.